pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,207 @@
+to the per-request contextor, to any implementation of the [Context](#context) object. 
+You can bind values, class definitions and provider functions to those contexts and 
+then resolve values (or the results of functions that return those values!) in other 
+areas of your code.
+
+```ts
+// application.ts
+import {Application} from '@loopback/core';
+import 'fs-extra';
+class MyApp extends Application {
+  constructor() {
+    super();
+    const app = this;
+    const widgetConf = JSON.parse(fs.readSync('./widget-config.json'));
+    app.bind('config.widget').to(widgetConf);
+  }
+}
+```
+Now that we've bound the 'config.widget' key to our configuration object, we can
+inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  constructor(
+    @inject('config.widget') protected widget: JSON
+    // This will be resolved at runtime!
+  ) {}
+  // etc...
+}
+```
+
+For more information, see the [Dependency Injection](Dependency-Injection.htm) section under [LoopBack Core Concepts](Concepts.htm)
+
+## Authentication Decorator
+
+- @authenticate
+
+  Mark a controller method as requiring authenticated user, takes a strategy name 
+  as a parameter. An example is using 'BasicStrategy' to authenticate user in function
+  `whoAmI`:
+
+  ```ts
+  // my-controller.ts
+  import { authenticate } from '@loopback/authentication';
+  import { inject } from '@loopback/context';
+
+  class MyController {
+    constructor(
+      @inject(BindingKeys.Authentication.CURRENT_USER)
+      private user: UserProfile,
+    ) {}
+
+    @authenticate('BasicStrategy')
+    async whoAmI() : Promise<string> {
+      return this.user.id;
+    }
+  }
+  ```
+
+## Repository Decorators
+
+### Model decorators
+
+- `@model`
+
+  Define a model in a repository, see [define-models](Repositories.html#define-models)
+
+- `@property`
+
+  Define a property in a model, see [define-models](Repositories.html#define-models)
+
+### Relation decorators
+
+  *This feature has not yet been released in alpha form. Documentation will be* 
+  *added here as this feature progresses.*
+
+- `@relation`
+
+  Register a general relation
+
+- `@belongsTo` `@hasOne` `@hasMany` `@embedsOne` `@embedsMany` `@referencesOne` `@referencesMany`
+
+  Register a specific relation
+
+### Repository decorator
+
+- `@repository`
+
+  A repository represents a collection of data and the means to access that data, 
+  the decorator either injects an existing repository or creates a repository 
+  from a model and a datasource.
+
+  The injection example can be found in [Repository#controller-configuration](Repositories.html#controller-configuration)
+
+  ```ts
+  // my-controller.ts
+  // Know more about how to create model 'Todo' and datasource 'datasource', 
+  // please see example in [Thinking in LoopBack](Thinking-in-LoopBack.htm#define-product-model-repository-and-data-source)
```

Comment:
Remove these comments.

Reply:
If the link is useful, it should be outside the code snippet, and properly formatted for use.